driver-async/src/main/com/mongodb/async/client/AbstractSubscription.java

```diff
@@ -134,9 +140,12 @@ void onError(final Throwable t) {
         }
     }
 
-    void onNext(final TResult next) {
+    private void onNext(final TResult next) {
         if (!isTerminated()) {
             try {
+                if (next == null) {
```

Comment:
Don't use exceptions for control flow.  As it is, it will end up logging this exception inaccurately when caught by the catch block (which was only intended to catch exceptions thrown by the observer.

Instead, move this out of the try block, and just call onError directly with the exception.

Reply:
Done